web3/_utils/method_formatters.py

```diff
@@ -580,14 +593,45 @@ def raise_transaction_not_found_with_index(params: Tuple[BlockIdentifier, int])
 }
 
 
+def filter_wrapper(
+    module: Union["Module", "ModuleV2"],
+    method: RPCEndpoint,
+    filter_id: HexStr,
+):
+    if method == RPC.eth_newBlockFilter:
+        return BlockFilter(filter_id, module=module)
+    elif method == RPC.eth_newPendingTransactionFilter:
+        return TransactionFilter(filter_id, module=module)
+    elif method == RPC.eth_newFilter:
+        return LogFilter(filter_id, module=module)
```

Comment:
I would add a final `else` clause here that raises an exception assuming you expect that logic branch to never be exercised.